metricbeat/helper/prometheus/textparse_test.go

```diff
@@ -0,0 +1,686 @@
+						SampleCount: uint64p(44000),
+						SampleSum:   float64p(234892394),
+						Quantile: []*Quantile{
+							{
+								Quantile: float64p(0.5),
+								Value:    float64p(29735),
+							},
+							{
+								Quantile: float64p(0.9),
+								Value:    float64p(47103),
+							},
+							{
+								Quantile: float64p(0.99),
+								Value:    float64p(50681),
+							},
+						},
+					},
+					TimestampMs: nil,
+				},
+			},
+		},
+	}
+
+	result, err := ParseMetricFamilies([]byte(input), ContentTypeTextFormat, time.Now())
+	if err != nil {
+		t.Fatalf("ParseMetricFamilies for content type %s returned an error.", ContentTypeTextFormat)
+	}
+	require.ElementsMatch(t, expected, result)
+}
+
+func TestHistogramOpenMetrics(t *testing.T) {
+	input := `
+# HELP http_server_requests_seconds Duration of HTTP server request handling
+# TYPE http_server_requests_seconds histogram
+http_server_requests_seconds{exception="None",uri="/actuator/prometheus",quantile="0.002796201"} 0.046137344
+http_server_requests_seconds{exception="None",uri="/actuator/prometheus",quantile="0.003145726"} 0.046137344
+http_server_requests_seconds{exception="None",uri="/actuator/prometheus",noquantile="0.005"} 1234
+http_server_requests_seconds_bucket{exception="None",uri="/actuator/prometheus",le="0.001"} 0.0
+http_server_requests_seconds_bucket{exception="None",uri="/actuator/prometheus",le="0.001048576"} 0.0
+http_server_requests_seconds_bucket{exception="None",uri="/actuator/prometheus",nole="0.002"} 1234
+http_server_requests_seconds_count{exception="None",uri="/actuator/prometheus"} 1.0
+http_server_requests_seconds_sum{exception="None",uri="/actuator/prometheus"} 0.046745444
+http_server_requests_seconds_created{exception="None",uri="/actuator/prometheus"} 0.046745444
+# EOF`
+	expected := []*MetricFamily{
+		{
+			Name: stringp("http_server_requests_seconds"),
+			Help: stringp("Duration of HTTP server request handling"),
+			Type: "histogram",
+			Unit: nil,
+			Metric: []*OpenMetric{
+				{
+					Label: []*labels.Label{
+						{
+							Name:  "exception",
+							Value: "None",
+						},
+						{
+							Name:  "uri",
+							Value: "/actuator/prometheus",
+						},
+					},
+					Name: stringp("http_server_requests_seconds"),
+					Histogram: &Histogram{
+						IsGaugeHistogram: false,
+						SampleCount:      uint64p(1.0),
+						SampleSum:        float64p(0.046745444),
+						Bucket: []*Bucket{
+							{
+								CumulativeCount: uint64p(0),
+								UpperBound:      float64p(0.001),
+							},
+							{
+								CumulativeCount: uint64p(0),
+								UpperBound:      float64p(0.001048576),
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	result, err := ParseMetricFamilies([]byte(input[1:]), OpenMetricsType, time.Now())
+	if err != nil {
+		t.Fatalf("ParseMetricFamilies for content type %s returned an error.", OpenMetricsType)
+	}
+	require.ElementsMatch(t, expected, result)
+}
+
+func TestHistogramPrometheus(t *testing.T) {
```

Comment:
Dummy question: Why we need both this test and TestHistogramOpenMetrics? 
When this will be called?

Reply:
For open metrics, we use a parser that is different than the one we use for prometheus metrics. That is decided by the content we give to the parse metrics function. So we are just checking the expected results for the two possible parsers: the `PromParser` and the `OpenMetricsParser`.